operator/api/v1alpha1/kyma_types.go

```diff
@@ -342,57 +322,50 @@ func init() {
 	SchemeBuilder.Register(&Kyma{}, &KymaList{})
 }
 
-const NewModuleMessage = "new module"
-
-func (kyma *Kyma) MatchConditionsToModules() {
-	for _, module := range kyma.Spec.Modules {
-		found := false
-		for _, condition := range kyma.Status.Conditions {
-			if module.Name == condition.Reason {
-				found = true
-			}
-		}
-		if !found {
-			newCondition := KymaCondition{
-				Type:    ConditionTypeReady,
-				Status:  ConditionStatusFalse,
-				Reason:  module.Name,
-				Message: NewModuleMessage,
-			}
-			kyma.Status.Conditions = append(kyma.Status.Conditions, newCondition)
+func (kyma *Kyma) InitModuleConditions() {
+	found := false
+	for _, condition := range kyma.Status.Conditions {
+		if condition.Reason == string(ConditionReasonModulesIsReady) {
+			found = true
 		}
 	}
+	if !found {
+		builder := NewConditionBuilder()
+		builder.SetStatus(metav1.ConditionFalse)
+		builder.SetReason(ConditionReasonModulesIsReady)
+		newCondition := builder.Build()
```

Comment:
why use a builder pattern if you anyhow have to use them all at once. Anyhow I think this is more functional Builder style so the right way would be
```suggestion
		newCondition := NewConditionBuilder().WithStatus(metav1.ConditionFalse).WithReason(ConditionReasonModulesIsReady)
```
formatted over multiple lines.

A much more common way in go to write builders is with varags like so:

```
BuildCondition(
  WithStatus(metav1.ConditionFalse),
  WithReason(ConditionReasonModulesIsReady)
)
```
pick your poison :)

Reply:
Use builder pattern here just want to reduce and encapsulate some condition params initialization, like `message`, `LastTransitionTime`.
Good, I take the suggestion to change the style to that chained style, I think the official name for this style is `Fluent interface`, but for that common way, I think the problem to me is lost the benefit to do verification in `build()` step, although I don't implement any yet:)

Reply:
But back to you first point, by the current state, I think it's more common change it to normal initialization, but maybe later iteration, some params like `type`, `message` will be exposed and can be optional, with this builder pattern is a bit flexible.